.circleci/config.yml

```diff
@@ -22,3 +22,30 @@ jobs:
             - ~/.m2
             - ~/workbench/api/build/exploded-api/WEB-INF/lib/
           key: api-cache-{{ checksum "~/workbench/api/build.gradle" }}
+  build_ui:
+    docker:
+        - image: circleci/node:latest-browsers
+    working_directory: ~/workbench
+    environment: []
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+          - ui-cache-{{ checksum "~/workbench/ui/package.json" }}
+          - ui-cache-
+      - run:
+          working_directory: ~/workbench/ui
+          command: npm install
+      - save_cache:
+          paths:
+            - ~/workbench/ui/node_modules
+          key: ui-cache-{{ checksum "~/workbench/ui/package.json" }}
+      - run:
+          working_directory: ~/workbench/ui
+          command: npm run test -- --no-watch
```

Comment:
karma (the thing that runs when `ng test` is invoked, as via `npm run test`) always tries to open browsers. The default behavior if you just do `ng test` (or `npm run test`) is to open a browser and keep it open, watching your files and re-running tests as you edit your code. Which is nice when you're debugging locally, but not great in CI (because it never finishes). Passing `--no-watch` runs the tests once and then closes out the browser (succeed or fail).